cli/mender-artifact/dump.go

```diff
@@ -178,67 +181,100 @@ func dumpMetaData(metaDataDir string, dumpArgs *[]string, handlers map[int]handl
 	return nil
 }
 
-func printCmdline(ar *areader.Reader, args []string) {
+type NullSepWriter struct {
+	buf *bytes.Buffer
+}
+
+func New() *NullSepWriter {
+	return &NullSepWriter{bytes.NewBuffer(nil)}
+}
+
+func (n *NullSepWriter) Write(b []byte) (int, error) {
+	n.buf.Write([]byte{'\x00'})
+	return n.buf.Write(b)
+}
+
+func (n *NullSepWriter) Read(b []byte) (int, error) {
+	return n.buf.Read(b)
+}
+
+func (n *NullSepWriter) String() string {
+	return n.buf.String()
+}
+
+type WriteStringer interface {
+	io.Writer
+	fmt.Stringer
+}
+
+func printCmdline(ar *areader.Reader, args []string, zeroSeparator bool) {
+	var buf WriteStringer
+	if zeroSeparator {
+		buf = New()
+	} else {
+		buf = bytes.NewBuffer(nil)
+	}
 	// Even if it is a rootfs payload, we use the module-image writer, since
 	// this can recreate either type.
-	fmt.Printf("write module-image")
+	fmt.Fprintf(buf, "write module-image")
```

Comment:
This also will not work, the separator has to be used *instead* of spaces, otherwise the spaces will become part of the argument. I was thinking more along the lines of:

```
sep := '\0'
fmt.Fprintf(buf, "write%cmodule-image", sep)
fmt.Fprintf(buf, "%c--artifact-name%c%s", sep, sep, artProvs.ArtifactName)
```

And so on.